ts/features/fims/store/reducers/index.ts

```diff
@@ -52,28 +56,90 @@ const reducer = (
         ...state,
         consentsData: pot.toError(state.consentsData, action.payload)
       };
+    case getType(fimsCancelOrAbortAction):
+      return pipe(
+        state.consentsData,
+        abortUrlFromConsentsPot,
+        O.fold(
+          () => ({ ...state }),
+          () => ({ ...state, currentFlowState: "abort" })
+        )
+      );
   }
   return state;
 };
 
 export const fimsConsentsDataSelector = (state: GlobalState) =>
   state.features.fims.consentsData;
 
+export const fimsPartialAbortUrl = (state: GlobalState) =>
+  pipe(state, fimsConsentsDataSelector, abortUrlFromConsentsPot, O.toUndefined);
+
+const abortUrlFromConsentsPot = (consentsPot: pot.Pot<ConsentData, string>) =>
+  pipe(
+    consentsPot,
+    foldK(
+      () => O.none,
+      () => O.none,
+      newConsents => O.some(newConsents),
+      _ => O.none,
+      consents => O.some(consents),
+      consents => O.some(consents),
+      (_, newConsents) => O.some(newConsents),
+      (consents, _) => O.some(consents)
+    ),
+    // eslint-disable-next-line no-underscore-dangle
+    O.map(consents => consents._links.abort.href)
```

Comment:
```suggestion
    pot.toOption,
    O.map(consents => consents._links.abort.href)
```

since we do not go through the `updating` pot state in our flow, this function has the same outcome as the one you used, but it comes pre-cooked!

Reply:
Nice catch, applied